docs/guide/database-insert-update.md

```diff
@@ -47,26 +53,34 @@ For example, if we had a dog adoption website we may want to update `dog` pieces
 **What is happening here?**
 
 ```javascript
-async registerInterest(dogId) {
-  const req = apos.task.getReq({ role: 'editor' });
-  const dogDocument = await self.find(req, { _id: dogId }).toObject();
+async updateDogStatus(req, dogId, status) {
+  const dogDocument = await self.find(req, {
+    dogId
+  }).toObject();
   ...
 }
 ```
 
-Our method received the `_id` of the document associated with this dog. We then generate a `req` object using `apos.task.getReq()`. While it is best to use a "natural" request object, if the contact form submission was from someone not logged in then the `req` will not have permission to update documents. We would want to make sure to guard against spam submissions elsewhere.
+Our method received the active request object, the API's identifier for a dog, and the dog's availability status (probably a string value). We then request the *full* database document. See the [database querying guide](/guide/database-queries.md) for more on this.
+
+```javascript
+if (!dogDocument) {
+  return null;
+};
+```
 
-We then request the *full* database document. See the [database querying guide](/guide/database-queries.md) for more on this.
+If the identified dog isn't in our database yet, we would return a `null` value and the update task's function could insert the dog instead.
 
 ```javascript
-dogDocument.receivedInterest++;
+// We update the `status` property.
+dogDocument.status = status;
 
 const updateResult = await self.update(req, dogDocument);
 
 return updateResult;
 ```
 
-We then update the document property that tracks how many times someone expressed interested in the dog. Then we use `self.update` to replace the previous document state with our update (with `await` as it is asynchronous). We finally return the result, which will be an object with update information from MongoDb.
+We update the document property that tracks the dog's adoption status on the data object and use `self.update` to replace the previous document state with our update (with `await` as it is asynchronous). We finally return the result, which will be an object with update information from MongoDb.
```

Comment:
The result of `self.update(req, doc, options)` will be the updated document itself, not a mongodb info object.